UNITTESTS/stubs/ATHandler_stub.cpp

```diff
@@ -74,28 +77,93 @@ ATHandler::~ATHandler()
 
 void ATHandler::inc_ref_count()
 {
-    ATHandler_stub::ref_count++;
+    _ref_count++;
+    ATHandler_stub::ref_count = _ref_count;
 }
 
 void ATHandler::dec_ref_count()
 {
-    ATHandler_stub::ref_count--;
+    _ref_count--;
+    ATHandler_stub::ref_count = _ref_count;
 }
 
 int ATHandler::get_ref_count()
 {
-    return ATHandler_stub::ref_count;
+    return _ref_count;
 }
 
 FileHandle *ATHandler::get_file_handle()
 {
-    return ATHandler_stub::fh_value;
+    ATHandler_stub::fh_value = (FileHandle_stub *)_fileHandle;
+    return _fileHandle;
 }
 
 void ATHandler::set_file_handle(FileHandle *fh)
 {
 }
 
+// each parser is associated with one filehandle (that is UART)
```

Comment:
Do you really need to paste two complete working routines into the stubs?

If you do, I suggest separating these two factory functions into a standalone ATHandler_factory.cpp, and including that working file in the tests, not copy-pasting the code.